### PR TITLE
Alternate for #1120: queueless worker silently fails new worker when being pruned

### PR DESCRIPTION
**Pull request by yaauie**
If a worker is all of:

1. registered
2. not listening on any queues
3. on our host
4. no longer running (via pid check)

It should get pruned.

See #1120 (cc: @dsabanin)

### CODE REVIEW

**Comment by chrisccerami** [comment]
@yaauie @steveklabnik Any thoughts on this PR at this point? I know it's been a long time. Can we move toward either merging or closing?

**Comment by steveklabnik** [comment]
I wanted to see if it was still needed or not, but haven't found the time. I think I merged an alternate PR to this?

**Comment by chrisccerami** [comment]
I can investigate at some point today and I'll post my findings here.

**Comment by steveklabnik** [comment]
@chrisccerami did you find anything relevant?

**Comment by chrisccerami** [comment]
Ugh, I either forgot to investigate or forgot to share what I found. I have a few free hours tonight and can try to look at it. My apologies.

**Comment by steveklabnik** [comment]
Zero worries. It's not a big deal :smile:

**Comment by chrisccerami** [comment]
@steveklabnik it seems like there may be an issue with this PR due to its age. If `1-x-stable` is merged/rebased back into this branch, the tests will fail as `Worker.all`/`Resque.workers` will fail with:

```
NoMethodError: undefined method `split' for nil:NilClass
    resque/lib/resque/worker.rb:93:in `find'
    resque/lib/resque/worker.rb:56:in `block in all'
    resque/lib/resque/worker.rb:56:in `map'
    resque/lib/resque/worker.rb:56:in `all'
    resque/lib/resque.rb:463:in `workers'
    resque/test/worker_test.rb:656:in `block (2 levels) in <top (required)>'
```

This wouldn't be too difficult to fix, you would just need to convert nil `to_s` or something similar, BUT there's another catch. If I modify the `find` method to not blow up when there's no queue, the `validates_queues` method catches it and raises a `NoQueueError`:

```
Resque::NoQueueError: Please give each worker at least one queue.
    resque/lib/resque/worker.rb:167:in `validate_queues'
    resque/lib/resque/worker.rb:158:in `queues='
    resque/lib/resque/worker.rb:149:in `initialize'
    resque/lib/resque/worker.rb:94:in `new'
    resque/lib/resque/worker.rb:94:in `find'
    resque/lib/resque/worker.rb:56:in `block in all'
    resque/lib/resque/worker.rb:56:in `map'
    resque/lib/resque/worker.rb:56:in `all'
    resque/lib/resque.rb:463:in `workers'
    resque/test/worker_test.rb:656:in `block (2 levels) in <top (required)>'
```

Basically, I'm not entirely convinced it's possible to get a Worker into a state such that it has no queues. This leads me to believe this work isn't really needed anymore.

**Comment by steveklabnik** [comment]
Awesome, thanks so much. I think I agree.
